lib/api/bgppeer.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	. "github.com/tigera/libcalico-go/lib/api/unversioned"
+	. "github.com/tigera/libcalico-go/lib/net"
+	"github.com/tigera/libcalico-go/lib/scope"
+)
+
+type BGPPeerMetadata struct {
+	ObjectMetadata
+
+	// The scope of the peer.  This may be global or node.  A global peer is a
+	// BGP device that peers with all Calico nodes.  A node peer is a BGP device that
+	// peers with the specified Calico node (specified by the node hostname).
+	Scope scope.GlobalOrNode `json:"scope" validate:"omitempty,scopeglobalornode"`
+
+	// The hostname of the node that is peering with this peer.  This is only valid
```

Comment:
"only valid" is a bit too weak for what we want, I think.  This field is `mandatory` for host scoped peers, and must not be defined for global scoped peers.